engine/api/workflow/run_workflow_test.go

```diff
@@ -367,7 +369,7 @@ func TestManualRun3(t *testing.T) {
 		}
 
 		//TakeNodeJobRun
-		j, _, err = workflow.TakeNodeJobRun(nil, db, db, cache, proj, j.ID, "model", "worker", "1", []sdk.SpawnInfo{
+		j, _, _ = workflow.TakeNodeJobRun(nil, func() *gorp.DbMap { return db }, db, cache, proj, j.ID, "model", "worker", "1", []sdk.SpawnInfo{
```

Comment:
do not pass a nil Context, even if a function permits it; pass context.TODO if you are unsure about which Context to use